metaci/api/views/testmethod_perf.py

```diff
@@ -0,0 +1,284 @@
+        / Cast(Count("id"), FloatField()),
+    }
+
+    metric_choices = tuple(zip(metrics.keys(), metrics.keys()))
+    include_fields = django_filters.rest_framework.MultipleChoiceFilter(
+        label="Include (multi-select okay)", choices=metric_choices, method=dummy
+    )
+
+    o = django_filters.rest_framework.OrderingFilter(
+        fields=(
+            metric_choices
+            + group_by_choices
+            + (
+                ("method_name", "method_name"),
+                ("count", "count"),
+                ("failures", "failures"),
+                ("assertion_failures", "assertion_failures"),
+                ("DML_failures", "DML_failures"),
+                ("Other_failures", "Other_failures"),
+                ("success_percentage", "success_percentage"),
+            )
+        )
+    )
+
+
+BUILD_FLOWS_LIMIT = 100
+
+
+class TestMethodPerfListView(generics.ListAPIView):
+    """
+    A view for lists of aggregated test metrics.
+
+    Note that the number of build flows covered is limited to **BUILD_FLOWS_LIMIT** for performance reasons. You can
+    change this default with the build_flows_limit parameter.
+    """
+
+    serializer_class = TestMethodPerfSerializer
+    filter_backends = (django_filters.rest_framework.DjangoFilterBackend,)
+    filter_class = TestMethodPerfFilter
+
+    # example URLs:
+    # http://localhost:8000/api/testmethod_perf/?repo=gem&plan=Release%20Test&method_name=testCreateNegative
+    # http://localhost:8000/api/testmethod_perf/?repo=Cumulus&plan=Feature%20Test&o=avg
+    # http://localhost:8000/api/testmethod_perf/?method_name=&repo=&plan=&flow=&recentdate=&daterange_after=&daterange_before=&o=-repo
+
+    def _get_buildflows(self):
+        """Which buildflows do we need to look at? Limit by time, repo, etc.
+           Also limit # for performance reasons"""
+        params = self.request.query_params
+        build_flows_limit = int(params.get("build_flows_limit") or BUILD_FLOWS_LIMIT)
+
+        buildflows = BuildFlow.objects.filter(tests_total__gte=1)
+        buildflows = BuildFlowFilterSet(
+            self.request.GET, buildflows, really_filter=True
+        ).qs
+
+        return buildflows.order_by("-time_end")[0:build_flows_limit]
+
+    def _get_aggregation_fields(self):
+        """What fields should appear in the output?"""
+        params = self.request.query_params
+        aggregations = {}
+        fields_to_include = params.getlist("include_fields")
+
+        if params.get("o"):
+            fields_to_include.append(params.get("o"))
+
+        fields = self.filter_class.metrics
+
+        for fieldname in fields_to_include:
+            fieldname = fieldname.strip("-")
+            aggregations[fieldname] = fields[fieldname]
+
+        print("Aggregating", aggregations)
+
+        return aggregations
+
+    def _get_splitter_fields(self):
+        "Which fields to split on (or group by, depending on how you think about it"
```

Comment:
Please use triple-quotes for consistency. Makes it more obvious at a glance that it's the docstring.